src/solvers/smt2_incremental/convert_expr_to_smt.cpp

```diff
@@ -1003,26 +1003,28 @@ static smt_termt convert_expr_to_smt(
 }
 
 static smt_termt convert_array_update_to_smt(
-  const exprt &old,
-  const exprt &index,
-  const exprt &new_value,
+  const with_exprt &with,
   const sub_expression_mapt &converted)
 {
-  const smt_termt &old_array_term = converted.at(old);
-  const smt_termt &index_term = converted.at(index);
-  const smt_termt &value_term = converted.at(new_value);
-  return smt_array_theoryt::store(old_array_term, index_term, value_term);
+  smt_termt array = converted.at(with.old());
+  auto it = ++with.operands().begin();
+  while(it != with.operands().end())
```

Comment:
⛏️ Why not changing to a `for` with iterator loop?

Reply:
Refactored.